lib/src/revset_graph.rs

```diff
@@ -163,12 +170,66 @@ where
             };
             self.nodes.insert(current_id.clone(), current_node);
             // Push to non-emitting list so the new head wouldn't be interleaved
-            self.new_head_ids.push(current_id);
+            self.new_head_ids.push_back(current_id);
         }
 
         Some(())
     }
 
+    /// Enqueues the first new head which will unblock the waiting ancestors.
+    ///
+    /// This does not move multiple head nodes to the queue because heads may be
+    /// connected to the fork points in arbitrary order.
```

Comment:
nit: should the function be called `flush_new_head()` then (singular)?

Reply:
fixed